docker-compose.yml

```diff
@@ -1,39 +1,59 @@
-version: '3.4'
+## Updated to work with the latest version of 
+## Foreman. Implemented fix for missing tables
+## in the database by running rake. 
+
+## To set the default admin user password run the
+## following after running: docker compose up -d
+## and waiting for all containers to become healthy
+##
+## docker exec container-name-here bundle exec bin/rake permissions:reset username=admin password=admin
+
+version: '3.5'
+name: foreman
+
 services:
   db:
+    image: postgres:17-alpine
+    restart: unless-stopped
+    hostname: db.example.com
     environment:
-      - POSTGRES_USER=foreman
       - POSTGRES_PASSWORD=foreman
-      - POSTGRES_DATABASE=foreman
-      - PGDATA=/var/lib/postgresql/data/pgdata
-    hostname: db.example.com
-    image: postgres:12
+      - PGDATA=/var/lib/postgresql/data
     ports:
       - '5432'
-    restart: always
+    networks:
+      - foreman_net
     healthcheck:
-      test: ["CMD-SHELL", "nc -z 127.0.0.1 5432 || exit 1"]
-      interval: 30s
-      timeout: 30s
+      test: ["CMD-SHELL", "pg_isready"]
+      interval: 10s
+      timeout: 5s
       retries: 3
     volumes:
       - db:/var/lib/postgresql/data
 
   app: &app_base
     image: quay.io/foreman/foreman:develop
-    command: bundle exec bin/rails server -b 0.0.0.0
+    networks:
+      - foreman_net
+    command: 
+      - /bin/bash
+      - -c
+      - |
+        bundle exec bin/rake db:reset
+        bundle exec bin/rake db:migrate
+        bundle exec bin/rails server -b 0.0.0.0
     build:
       context: .
     environment:
-      - DATABASE_URL=postgres://foreman:foreman@db/foreman?pool=5
+      - DATABASE_URL=postgres://postgres:foreman@db:5432/foreman?pool=5
       - RAILS_MAX_THREADS=5
       - RAILS_ENV=production
       - FOREMAN_FQDN=foreman.example.com
       - FOREMAN_DOMAIN=example.com
       - FOREMAN_RAILS_CACHE_STORE_TYPE=redis
       - FOREMAN_RAILS_CACHE_STORE_URLS=redis://redis-cache:6379/0
       - DYNFLOW_REDIS_URL=redis://redis-tasks:6379/0
-      - REDIS_PROVIDER=DYNFLOW_REDIS_URL
+      - REDIS_PROVIDER=$DYNFLOW_REDIS_URL
```

Comment:
I don't think this is right. Looking at https://github.com/sidekiq/sidekiq/blob/77c6052e4d2d9ccc26457b4a2d629c33a9456664/lib/sidekiq/redis_connection.rb#L93-L98 it should be the name of the env var that contains the real URL.